picoquic/tls_api.c

```diff
@@ -1230,23 +1230,23 @@ typedef struct {
 typedef struct {
     /* The pointer to the overlying `verify_ctx` */
     void *verify_ctx;
-    int (*verify_sign)(void *verify_ctx, ptls_iovec_t data, ptls_iovec_t sign);
+    int (*verify_sign)(void *verify_ctx, uint16_t algo, ptls_iovec_t data, ptls_iovec_t sign);
 } picoquic_verify_ctx_t;
 
```

Comment:
Don't you love Kazuho's macros? The callback structure `ptls_verify_certificate_t` is defined as 
```
PTLS_CALLBACK_TYPE(int, verify_certificate, ptls_t *tls,
                   int (**verify_sign)(void *verify_ctx, ptls_iovec_t data, ptls_iovec_t sign), void **verify_data,
                   ptls_iovec_t *certs, size_t num_certs);
```
in `picotls.h`. I assume that you changed that, I saw that Kazuho merged your changes. But then, your code depends from this latest commit in the `picotls` repo.

The `picoquic` builds call the scripts `./ci/build_picotls.sh` (for Unix) and `./ci/build_picotls.ps1` (for Windows), which contain code like:
```
# Build at a known-good commit
COMMIT_ID=d5e32519516e13520ddeffa85da97d2398197df2
```
You want to update that to a `picotls` commit number posterior to your changes, that should clear the failures of the check-in tests.

Reply:
Specifically, in file `(repo)/ci/build_picotls.h`, replace:
```
# Build at a known-good commit
COMMIT_ID=d5e32519516e13520ddeffa85da97d2398197df2
```
By:
```
# Build at a known-good commit
COMMIT_ID=c91baf9c9971332d9280a7062943c54a4f13a7ec
```
And in file `(repo)/ci/build_picotls.ps1`, replace:
```
# Build at a known-good commit
$COMMIT_ID="d5e32519516e13520ddeffa85da97d2398197df2"
```
by:
```
# Build at a known-good commit
$COMMIT_ID="c91baf9c9971332d9280a7062943c54a4f13a7ec"
```